File: zk_auth_lib/go/chaum_pedersen_test.go
package chaum_pedersen_test

import (
	"math/big"
	"math/rand"
	"testing"

	cp "zk_auth_lib"
)

func TestChaumPedersen(t *testing.T) {
	q := big.NewInt(10003)
	g := big.NewInt(3)
	a := big.NewInt(10)
	b := big.NewInt(13)
	p := cp.NewChaumPedersen(g, q, a, b)
	v := cp.NewChaumPedersen(g, q, a, b)
	r := big.NewInt(rand.Int63n(1000))

	y1, y2 := p.Register(r)
	s := v.Challenge()
	z := p.Response(s)
	if !v.Verify(y1, y2, z) {
		t.Error("Failed to verify")
	}
}

File: zk_auth_lib/py/chaum_pedersen_test.py
#!/usr/bin/env python3.7
# coding=utf-8

import random
import unittest

import logging
import sys
import os
LOGGER = logging.getLogger()

from chaum_pedersen import ChaumPedersenData, ChaumPedersenProver, ChaumPedersenVerifier



class ChaumPedersenTest(unittest.TestCase):
    def __init__(self, methodName: str = "runTest") -> None:
        super().__init__(methodName)
        logging.basicConfig(
            level=logging.DEBUG if os.getenv('DEBUG') == 'True' else logging.INFO,
            format="%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s",
            handlers=[
                logging.StreamHandler(sys.stdout)
            ])
        LOGGER.info("ChaumPedersenTest")


    def test_functional(self):        
        T = type("", (), {})()
        T.q = 10003
        T.g = 3
        T.a = 10
        T.b = 13
        p = ChaumPedersenProver(T)

        # r is not shared with the prover on initialization
        r = random.randint(1,1000)
        T.r = r
        v = ChaumPedersenVerifier(T)

        p.register(r)
        s = v.challenge()
        z = p.response(s)
        res = v.verify(z)
        
        self.assertEqual(res, True)


